# cpDNA

[TOC levels=1-3]: # " "
- [cpDNA](#cpdna)
- [*Medicago truncatula*](#medicago-truncatula)
- [HM050](#hm050)
    - [HM050: download](#hm050-download)
    - [HM050: template](#hm050-template)
    - [HM050: run](#hm050-run)
- [HM340](#hm340)
    - [HM340: download](#hm340-download)
    - [HM340: template](#hm340-template)
    - [HM340: run](#hm340-run)


* Rsync to hpcc

```bash
rsync -avP \
    ~/data/dna-seq/cpDNA/Medicago/ \
    wangq@202.119.37.251:data/dna-seq/cpDNA/Medicago

# rsync -avP wangq@202.119.37.251:data/dna-seq/cpDNA/ ~/data/dna-seq/cpDNA

```

# *Medicago truncatula*

* Genome: GCF_000219495.3, 389.98 Mb
* Chloroplast: NC_003119, 124033 bp
* Mitochondrion: NC_029641, 271618 bp

* Reference genome

```bash
mkdir -p ~/data/dna-seq/cpDNA/Medicago
cd ~/data/dna-seq/cpDNA/Medicago

for ACCESSION in "NC_003119" "NC_029641"; do
    URL=$(printf "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&rettype=%s&id=%s&retmode=text" "fasta" "${ACCESSION}");
    curl $URL -o ${ACCESSION}.fa
done

aria2c -x 9 -s 3 -c ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/219/495/GCF_000219495.3_MedtrA17_4.0/GCF_000219495.3_MedtrA17_4.0_genomic.fna.gz

TAB=$'\t'
cat <<EOF > replace.tsv
NC_016407.2${TAB}1
NC_016408.2${TAB}2
NC_016409.2${TAB}3
NC_016410.2${TAB}4
NC_016411.2${TAB}5
NC_016412.2${TAB}6
NC_016413.2${TAB}7
NC_016414.2${TAB}8
NC_003119.8${TAB}Pt
NC_029641.1${TAB}Mt
EOF

cat NC_003119.fa NC_029641.fa |
    faops replace stdin replace.tsv stdout |
    faops order stdin <(echo Pt; echo Mt) genome.fa

faops replace GCF_000219495.3*_genomic.fna.gz replace.tsv stdout |
    faops order stdin <(echo {1..8}) ref.fa

```

* Illumina

```bash
cd ~/data/dna-seq/cpDNA/Medicago

cat << EOF > sra_ftp.txt
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/008/SRR1552478/SRR1552478_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/008/SRR1552478/SRR1552478_2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR152/005/SRR1524305/SRR1524305_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR152/005/SRR1524305/SRR1524305_2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR103/003/SRR1034293/SRR1034293_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR103/003/SRR1034293/SRR1034293_2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR216/006/SRR2163426/SRR2163426_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR216/006/SRR2163426/SRR2163426_2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR166/008/SRR1664358/SRR1664358_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR166/008/SRR1664358/SRR1664358_2.fastq.gz
EOF

aria2c -x 9 -s 3 -c -i sra_ftp.txt

cat << EOF > sra_md5.txt
3f570ce4060ffcb741f43501b3afe94f  SRR1552478_1.fastq.gz
b45ce0743d9f446e62473963ce7ac40b  SRR1552478_2.fastq.gz
f03019da3054b8659b80ffdaf388599f  SRR1524305_1.fastq.gz
ff624191ea6080c87663ad78e844ac80  SRR1524305_2.fastq.gz
52581dcbcb9506dffe3d920fc0d83921  SRR1034293_1.fastq.gz
3582cc84b309658811259d38d514cf62  SRR1034293_2.fastq.gz
28ce4e3b8138df654780401245113bf9  SRR2163426_1.fastq.gz
8756862c6043ffa96a8b1a5413865cd2  SRR2163426_2.fastq.gz
a86d4a063e107f2ce77ef1ca0051df86  SRR1664358_1.fastq.gz
5b6ec25a47d48194b7afea6bbaa6a6a3  SRR1664358_2.fastq.gz
EOF

md5sum --check sra_md5.txt

```

# HM050

*Medicago truncatula* HM050

## HM050: download

* Illumina
    * [SRR1034293](https://www.ncbi.nlm.nih.gov/sra/?term=SRR1034293)
    * 22.7 Gb
    * ~58.2x

```bash
mkdir -p ~/data/dna-seq/cpDNA/HM050/1_genome
cd ~/data/dna-seq/cpDNA/HM050/1_genome

ln -fs ../../Medicago/genome.fa genome.fa
ln -fs ../../Medicago/ref.fa ref.fa

mkdir -p ~/data/dna-seq/cpDNA/HM050/2_illumina
cd ~/data/dna-seq/cpDNA/HM050/2_illumina

ln -fs ~/data/dna-seq/cpDNA/Medicago/SRR1034293_1.fastq.gz R1.fq.gz
ln -fs ~/data/dna-seq/cpDNA/Medicago/SRR1034293_2.fastq.gz R2.fq.gz

```


## HM050: template

`--matchk 27 --artifact ../../1_genome/ref.fa`: doesn't work

```bash
WORKING_DIR=${HOME}/data/dna-seq/cpDNA
BASE_NAME=HM050

cd ${WORKING_DIR}/${BASE_NAME}

rm -f *.sh
anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1000000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 120 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160" \
    --tadpole \
    --splitp 100 \
    --statp 2 \
    --fillanchor \
    --redoanchors \
    --parallel 24 \
    --xmx 110g

```

## HM050: run

```bash
WORKING_DIR=${HOME}/data/dna-seq/cpDNA
BASE_NAME=HM050

cd ${WORKING_DIR}/${BASE_NAME}
#rm -fr 4_*/ 6_*/ 7_*/ 8_*/ 2_illumina/trim 2_illumina/mergereads statReads.md 

bash 0_bsub.sh
#bkill -J "${BASE_NAME}-*"

#bash 0_master.sh
#bash 0_cleanup.sh

```

Table: statInsertSize

| Group           |  Mean | Median | STDev | PercentOfPairs/PairOrientation |
|:----------------|------:|-------:|------:|-------------------------------:|
| tadpole.bbtools | 327.5 |    356 |  79.2 |                         15.64% |
| tadpole.picard  | 329.7 |    356 |  75.3 |                             FR |


Table: statReads

| Name     | N50 |    Sum |         # |
|:---------|----:|-------:|----------:|
| Illumina | 100 | 22.71G | 227105112 |
| trim     | 100 |  8.47G |  87657228 |
| Q25L60   | 100 |  7.86G |  82222188 |


Table: statTrimReads

| Name     | N50 |    Sum |         # |
|:---------|----:|-------:|----------:|
| clumpify | 100 | 20.52G | 205232542 |
| highpass | 100 |  9.85G |  98548676 |
| trim     | 100 |  8.47G |  87657236 |
| filter   | 100 |  8.47G |  87657228 |
| R1       | 100 |  4.23G |  43828614 |
| R2       | 100 |  4.25G |  43828614 |
| Rs       |   0 |      0 |         0 |


```text
#trim
#Matched	463084	0.46990%
#Name	Reads	ReadsPct
TruSeq_Adapter_Index_1_6	150051	0.15226%
Nextera_LMP_Read2_External_Adapter	98683	0.10014%
PhiX_read1_adapter	59497	0.06037%
pcr_dimer	58733	0.05960%
Reverse_adapter	58271	0.05913%
PCR_Primers	12001	0.01218%
TruSeq_Universal_Adapter	8626	0.00875%
PhiX_read2_adapter	3955	0.00401%
Nextera_LMP_Read1_External_Adapter	2699	0.00274%
RNA_PCR_Primer_Index_1_(RPI1)_2,9	1618	0.00164%
RNA_Adapter_(RA5)_part_#_15013205	1498	0.00152%
I7_Nextera_Transposase_1	1202	0.00122%
I5_Nextera_Transposase_1	1048	0.00106%
```

```text
#filter
#Matched	4	0.00000%
#Name	Reads	ReadsPct
```

```text
#peaks.raw
#k	31
#unique_kmers	1445271598
#main_peak	12
#genome_size	316563643
#haploid_genome_size	316563643
#fold_coverage	12
#haploid_fold_coverage	12
#ploidy	1
#percent_repeat	27.396
#start	center	stop	max	volume
7	12	30	12379538	229838259	
443	445	473	2710	76216	
473	475	476	2435	7234	
476	478	503	2383	61488	
503	504	505	2198	4279	
505	506	530	2149	51311	
530	531	574	2019	82171	
574	576	594	1792	34667	
594	595	628	1699	55160	
628	634	655	1564	41007	
655	670	672	1456	24478	
672	673	781	1461	138966	
781	885	969	1119	195922	
969	995	1071	900	82703	
1071	1077	2693	734	423282	
```

```text
#peaks.highpass
#k	31
#unique_kmers	337332955
#main_peak	126
#genome_size	9104454
#haploid_genome_size	9104454
#fold_coverage	126
#haploid_fold_coverage	126
#ploidy	1
#percent_repeat	74.272
#start	center	stop	max	volume
122	126	315	26596	2342415	
524	527	529	1841	9123	
529	532	569	1817	68968	
569	576	616	1743	73189	
616	626	655	1508	55642	
655	656	698	1368	56448	
698	708	768	1327	82964	
768	795	809	1135	45176	
809	813	814	1089	5328	
814	816	841	1050	28085	
841	872	881	1059	40337	
881	883	1018	1067	121925	
1018	1020	2531	790	427266	
```


Table: statMergeReads

| Name          | N50 |   Sum |        # |
|:--------------|----:|------:|---------:|
| clumped       | 100 | 8.45G | 87437634 |
| ecco          | 100 | 8.44G | 87437634 |
| eccc          | 100 | 8.44G | 87437634 |
| ecct          | 100 | 6.49G | 67417104 |
| extended      | 130 | 8.38G | 67417104 |
| merged        | 390 | 3.78G | 13122967 |
| unmerged.raw  | 125 | 5.03G | 41171170 |
| unmerged.trim | 125 | 5.03G | 41161190 |
| U1            | 125 | 2.51G | 20580595 |
| U2            | 125 | 2.53G | 20580595 |
| Us            |   0 |     0 |        0 |
| pe.cor        | 140 | 8.83G | 67407124 |

| Group            |  Mean | Median | STDev | PercentOfPairs |
|:-----------------|------:|-------:|------:|---------------:|
| ihist.merge1.txt | 144.4 |    156 |  36.2 |         10.78% |
| ihist.merge.txt  | 288.3 |    368 | 121.8 |         38.93% |


Table: statQuorum

| Name   |  CovIn | CovOut | Discard% | AvgRead | Kmer | RealG |   EstG | Est/Real |   RunTime |
|:-------|-------:|-------:|---------:|--------:|-----:|------:|-------:|---------:|----------:|
| Q0L0   | 8473.2 | 7877.9 |    7.03% |      97 | "69" |    1M | 49.48M |    49.48 | 0:14'43'' |
| Q25L60 | 7863.7 | 7394.9 |    5.96% |      96 | "69" |    1M | 45.72M |    45.72 | 0:13'40'' |


Table: statKunitigsAnchors.md

| Name           | CovCor | Mapped% | N50Anchor |     Sum |   # | N50Others |     Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeKU | RunTimeAN |
|:---------------|-------:|--------:|----------:|--------:|----:|----------:|--------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000    |   40.0 |  10.78% |      1632 | 119.22K |  73 |      1792 |  108.2K | 128 |   13.0 | 7.0 |   3.0 |  26.0 | "31,41,51,61,71,81" | 0:00'16'' | 0:00'28'' |
| Q0L0X40P001    |   40.0 |  10.96% |      1580 | 114.04K |  73 |      1591 | 124.01K | 142 |   13.0 | 7.0 |   3.0 |  26.0 | "31,41,51,61,71,81" | 0:00'15'' | 0:00'28'' |
| Q0L0X40P002    |   40.0 |  11.05% |      1718 | 112.74K |  68 |      1738 | 138.32K | 140 |   12.0 | 6.0 |   3.0 |  24.0 | "31,41,51,61,71,81" | 0:00'15'' | 0:00'28'' |
| Q0L0X80P000    |   80.0 |  10.76% |      2203 | 217.51K | 108 |      1843 | 142.94K | 163 |   14.0 | 5.0 |   3.0 |  28.0 | "31,41,51,61,71,81" | 0:00'25'' | 0:00'28'' |
| Q0L0X80P001    |   80.0 |  12.00% |      2082 | 216.43K | 106 |      1904 | 160.86K | 160 |   14.0 | 5.0 |   3.0 |  28.0 | "31,41,51,61,71,81" | 0:00'25'' | 0:00'29'' |
| Q0L0X80P002    |   80.0 |  11.62% |      2441 | 210.58K |  99 |      1539 | 168.59K | 176 |   14.0 | 5.0 |   3.0 |  28.0 | "31,41,51,61,71,81" | 0:00'25'' | 0:00'29'' |
| Q0L0X120P000   |  120.0 |   9.67% |      2062 | 226.88K | 112 |      1761 | 141.49K | 114 |   18.0 | 5.0 |   3.0 |  36.0 | "31,41,51,61,71,81" | 0:00'35'' | 0:00'30'' |
| Q0L0X120P001   |  120.0 |  10.28% |      2613 | 236.13K | 106 |      1628 | 137.75K | 150 |   19.0 | 5.0 |   3.0 |  38.0 | "31,41,51,61,71,81" | 0:00'35'' | 0:00'30'' |
| Q0L0X120P002   |  120.0 |  10.20% |      2242 | 237.48K | 115 |      1544 | 141.63K | 149 |   18.0 | 5.0 |   3.0 |  36.0 | "31,41,51,61,71,81" | 0:00'35'' | 0:00'30'' |
| Q0L0X160P000   |  160.0 |   9.05% |      2026 |  232.4K | 118 |      1641 | 124.95K | 112 |   23.0 | 6.0 |   3.0 |  46.0 | "31,41,51,61,71,81" | 0:00'45'' | 0:00'31'' |
| Q0L0X160P001   |  160.0 |   8.50% |      1828 |  222.6K | 121 |      1621 | 122.41K | 107 |   24.0 | 6.0 |   3.0 |  48.0 | "31,41,51,61,71,81" | 0:00'44'' | 0:00'30'' |
| Q0L0X160P002   |  160.0 |   9.16% |      2048 | 218.04K | 111 |      1628 | 132.52K | 111 |   24.0 | 7.0 |   3.0 |  48.0 | "31,41,51,61,71,81" | 0:00'44'' | 0:00'30'' |
| Q25L60X40P000  |   40.0 |  13.24% |      1664 | 131.17K |  82 |      1719 |  138.4K | 162 |   13.0 | 7.0 |   3.0 |  26.0 | "31,41,51,61,71,81" | 0:00'15'' | 0:00'28'' |
| Q25L60X40P001  |   40.0 |  11.26% |      1434 |  117.7K |  81 |      1860 | 129.01K | 138 |   12.0 | 6.0 |   3.0 |  24.0 | "31,41,51,61,71,81" | 0:00'15'' | 0:00'28'' |
| Q25L60X40P002  |   40.0 |  12.31% |      1532 | 144.56K |  93 |      1793 | 118.55K | 145 |   13.0 | 7.0 |   3.0 |  26.0 | "31,41,51,61,71,81" | 0:00'15'' | 0:00'27'' |
| Q25L60X80P000  |   80.0 |  13.78% |      2540 | 224.36K |  99 |      1959 | 183.25K | 171 |   15.0 | 6.0 |   3.0 |  30.0 | "31,41,51,61,71,81" | 0:00'25'' | 0:00'29'' |
| Q25L60X80P001  |   80.0 |  13.42% |      2350 | 215.09K | 101 |      1796 | 190.67K | 176 |   15.0 | 6.0 |   3.0 |  30.0 | "31,41,51,61,71,81" | 0:00'25'' | 0:00'29'' |
| Q25L60X80P002  |   80.0 |  12.24% |      2148 | 211.21K | 106 |      1819 | 163.28K | 164 |   14.0 | 5.0 |   3.0 |  28.0 | "31,41,51,61,71,81" | 0:00'25'' | 0:00'29'' |
| Q25L60X120P000 |  120.0 |  12.44% |      2773 | 236.67K |  99 |      1783 | 170.52K | 145 |   20.0 | 6.5 |   3.0 |  40.0 | "31,41,51,61,71,81" | 0:00'34'' | 0:00'30'' |
| Q25L60X120P001 |  120.0 |  10.86% |      2160 | 217.15K | 106 |      1739 | 152.72K | 123 |   19.0 | 6.0 |   3.0 |  38.0 | "31,41,51,61,71,81" | 0:00'34'' | 0:00'29'' |
| Q25L60X120P002 |  120.0 |  10.75% |      2408 | 232.92K | 107 |      1613 | 149.11K | 154 |   19.0 | 5.0 |   3.0 |  38.0 | "31,41,51,61,71,81" | 0:00'35'' | 0:00'29'' |
| Q25L60X160P000 |  160.0 |  10.17% |      2261 | 229.13K | 110 |      1627 | 142.69K | 118 |   25.0 | 7.0 |   3.0 |  50.0 | "31,41,51,61,71,81" | 0:00'43'' | 0:00'30'' |
| Q25L60X160P001 |  160.0 |   8.75% |      2087 | 222.68K | 112 |      1345 | 119.43K | 112 |   24.0 | 6.0 |   3.0 |  48.0 | "31,41,51,61,71,81" | 0:00'44'' | 0:00'30'' |
| Q25L60X160P002 |  160.0 |   9.99% |      2286 | 231.14K | 109 |      1750 | 130.46K | 108 |   24.0 | 6.0 |   3.0 |  48.0 | "31,41,51,61,71,81" | 0:00'43'' | 0:00'30'' |


Table: statTadpoleAnchors.md

| Name           | CovCor | Mapped% | N50Anchor |     Sum |   # | N50Others |     Sum |   # | median |  MAD | lower | upper |                Kmer | RunTimeKU | RunTimeAN |
|:---------------|-------:|--------:|----------:|--------:|----:|----------:|--------:|----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000    |   40.0 |  13.86% |      2383 |  99.47K |  46 |      2483 |  44.51K |  42 |   21.0 | 10.0 |   3.0 |  42.0 | "31,41,51,61,71,81" | 0:00'26'' | 0:00'28'' |
| Q0L0X40P001    |   40.0 |  15.71% |      2025 |  96.47K |  48 |      1600 |  64.53K |  64 |   21.0 | 10.5 |   3.0 |  42.0 | "31,41,51,61,71,81" | 0:00'21'' | 0:00'28'' |
| Q0L0X40P002    |   40.0 |  16.11% |      2190 | 107.09K |  53 |      1940 |  60.22K |  68 |   21.0 | 10.5 |   3.0 |  42.0 | "31,41,51,61,71,81" | 0:00'24'' | 0:00'27'' |
| Q0L0X80P000    |   80.0 |  16.53% |      2149 | 208.03K | 103 |      3423 | 179.72K | 112 |   17.0 |  7.0 |   3.0 |  34.0 | "31,41,51,61,71,81" | 0:00'31'' | 0:00'29'' |
| Q0L0X80P001    |   80.0 |  17.92% |      1934 | 199.79K | 103 |      3620 | 182.84K | 103 |   17.0 |  7.0 |   3.0 |  34.0 | "31,41,51,61,71,81" | 0:00'30'' | 0:00'30'' |
| Q0L0X80P002    |   80.0 |  17.12% |      2100 | 200.39K | 103 |      3762 | 174.94K | 107 |   18.0 |  8.0 |   3.0 |  36.0 | "31,41,51,61,71,81" | 0:00'29'' | 0:00'28'' |
| Q0L0X120P000   |  120.0 |  20.76% |      3335 | 259.92K |  90 |      4027 | 220.19K | 101 |   23.0 |  8.0 |   3.0 |  46.0 | "31,41,51,61,71,81" | 0:00'33'' | 0:00'30'' |
| Q0L0X120P001   |  120.0 |  20.22% |      3941 | 261.97K |  87 |      4008 | 215.03K | 118 |   21.0 |  7.0 |   3.0 |  42.0 | "31,41,51,61,71,81" | 0:00'33'' | 0:00'30'' |
| Q0L0X120P002   |  120.0 |  20.50% |      4146 | 263.65K |  84 |      3811 | 231.56K | 119 |   22.0 |  8.0 |   3.0 |  44.0 | "31,41,51,61,71,81" | 0:00'32'' | 0:00'31'' |
| Q0L0X160P000   |  160.0 |  20.30% |      4630 | 285.36K |  77 |      4492 |    221K |  98 |   27.0 |  9.0 |   3.0 |  54.0 | "31,41,51,61,71,81" | 0:00'36'' | 0:00'30'' |
| Q0L0X160P001   |  160.0 |  20.86% |      6609 | 285.17K |  77 |      3515 | 233.82K | 107 |   27.0 |  9.5 |   3.0 |  54.0 | "31,41,51,61,71,81" | 0:00'37'' | 0:00'31'' |
| Q0L0X160P002   |  160.0 |  20.51% |      6320 | 280.27K |  67 |      4043 | 219.27K | 101 |   27.0 |  8.5 |   3.0 |  54.0 | "31,41,51,61,71,81" | 0:00'37'' | 0:00'32'' |
| Q25L60X40P000  |   40.0 |  18.97% |      1985 | 102.12K |  48 |      1997 |   57.4K |  59 |   23.0 | 11.0 |   3.0 |  46.0 | "31,41,51,61,71,81" | 0:00'22'' | 0:00'27'' |
| Q25L60X40P001  |   40.0 |  15.97% |      1855 | 101.89K |  52 |      2472 |  53.83K |  51 |   21.0 |  9.0 |   3.0 |  42.0 | "31,41,51,61,71,81" | 0:00'24'' | 0:00'27'' |
| Q25L60X40P002  |   40.0 |  18.77% |      1880 | 106.02K |  51 |      2319 |     56K |  51 |   23.0 | 10.0 |   3.0 |  46.0 | "31,41,51,61,71,81" | 0:00'21'' | 0:00'28'' |
| Q25L60X80P000  |   80.0 |  21.49% |      1935 | 205.43K | 109 |      4178 | 189.69K | 103 |   19.0 |  8.0 |   3.0 |  38.0 | "31,41,51,61,71,81" | 0:00'30'' | 0:00'29'' |
| Q25L60X80P001  |   80.0 |  21.24% |      2142 | 195.43K |  97 |      4274 | 183.77K | 109 |   18.0 |  7.0 |   3.0 |  36.0 | "31,41,51,61,71,81" | 0:00'29'' | 0:00'29'' |
| Q25L60X80P002  |   80.0 |  22.25% |      2023 | 207.45K | 105 |      4043 | 173.79K | 101 |   20.0 |  9.0 |   3.0 |  40.0 | "31,41,51,61,71,81" | 0:00'28'' | 0:00'30'' |
| Q25L60X120P000 |  120.0 |  23.15% |      4219 | 263.43K |  83 |      4464 | 213.56K | 108 |   22.0 |  7.5 |   3.0 |  44.0 | "31,41,51,61,71,81" | 0:00'31'' | 0:00'32'' |
| Q25L60X120P001 |  120.0 |  23.76% |      3134 | 264.56K | 101 |      4130 | 225.85K | 104 |   23.0 |  8.0 |   3.0 |  46.0 | "31,41,51,61,71,81" | 0:00'33'' | 0:00'31'' |
| Q25L60X120P002 |  120.0 |  20.88% |      3421 | 260.54K |  89 |      4052 | 197.17K |  92 |   23.0 |  9.0 |   3.0 |  46.0 | "31,41,51,61,71,81" | 0:00'33'' | 0:00'30'' |
| Q25L60X160P000 |  160.0 |  22.49% |      6318 | 275.02K |  67 |      4190 |    216K |  94 |   27.0 |  8.0 |   3.0 |  54.0 | "31,41,51,61,71,81" | 0:00'36'' | 0:00'31'' |
| Q25L60X160P001 |  160.0 |  22.44% |      7518 | 275.41K |  69 |      4181 |  198.1K |  83 |   27.0 |  8.0 |   3.0 |  54.0 | "31,41,51,61,71,81" | 0:00'36'' | 0:00'31'' |
| Q25L60X160P002 |  160.0 |  23.29% |      6211 | 280.73K |  74 |      4492 | 221.76K |  96 |   27.0 |  8.0 |   3.0 |  54.0 | "31,41,51,61,71,81" | 0:00'36'' | 0:00'32'' |


Table: statMRKunitigsAnchors.md

| Name | CovCor | Mapped% | N50Anchor | Sum | # | N50Others | Sum | # | median | MAD | lower | upper | Kmer | RunTimeKU | RunTimeAN |
|:-----|-------:|--------:|----------:|----:|--:|----------:|----:|--:|-------:|----:|------:|------:|-----:|----------:|----------:|


Table: statMRTadpoleAnchors.md

| Name       | CovCor | Mapped% | N50Anchor |     Sum |  # | N50Others |     Sum |   # | median |  MAD | lower | upper |                Kmer | RunTimeKU | RunTimeAN |
|:-----------|-------:|--------:|----------:|--------:|---:|----------:|--------:|----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000  |   40.0 |  20.61% |      7215 | 264.88K | 54 |      6202 | 178.56K | 150 |   12.0 |  3.0 |   3.0 |  24.0 | "31,41,51,61,71,81" | 0:00'24'' | 0:00'31'' |
| MRX40P001  |   40.0 |  20.74% |      5856 |  255.4K | 64 |      4577 | 186.53K | 170 |   12.0 |  4.0 |   3.0 |  24.0 | "31,41,51,61,71,81" | 0:00'24'' | 0:00'30'' |
| MRX40P002  |   40.0 |  19.81% |      6470 | 263.85K | 55 |      4548 | 158.73K | 138 |   13.0 |  5.0 |   3.0 |  26.0 | "31,41,51,61,71,81" | 0:00'25'' | 0:00'29'' |
| MRX80P000  |   80.0 |  21.52% |     22462 | 290.99K | 36 |      7063 | 184.78K |  84 |   23.0 | 12.0 |   3.0 |  46.0 | "31,41,51,61,71,81" | 0:00'31'' | 0:00'29'' |
| MRX80P001  |   80.0 |  21.53% |     26260 | 289.07K | 31 |      6584 |  181.4K |  85 |   24.0 | 18.0 |   3.0 |  48.0 | "31,41,51,61,71,81" | 0:00'32'' | 0:00'29'' |
| MRX80P002  |   80.0 |  21.98% |     22680 | 282.77K | 24 |      5810 | 170.93K |  79 |   23.0 | 11.5 |   3.0 |  46.0 | "31,41,51,61,71,81" | 0:00'31'' | 0:00'31'' |
| MRX120P000 |  120.0 |  22.00% |     22727 | 299.92K | 42 |      6330 |  203.9K |  92 |   34.0 | 28.0 |   3.0 |  68.0 | "31,41,51,61,71,81" | 0:00'35'' | 0:00'30'' |
| MRX120P001 |  120.0 |  19.77% |     21093 |  295.2K | 38 |      5919 | 159.23K |  73 |   34.0 | 28.0 |   3.0 |  68.0 | "31,41,51,61,71,81" | 0:00'37'' | 0:00'30'' |
| MRX120P002 |  120.0 |  19.96% |     20702 | 298.94K | 39 |      6025 | 159.57K |  76 |   34.0 | 27.5 |   3.0 |  68.0 | "31,41,51,61,71,81" | 0:00'35'' | 0:00'30'' |
| MRX160P000 |  160.0 |  19.48% |     22727 | 298.93K | 44 |      5524 | 164.31K |  80 |   43.0 | 36.0 |   3.0 |  86.0 | "31,41,51,61,71,81" | 0:00'37'' | 0:00'30'' |
| MRX160P001 |  160.0 |  19.42% |     19953 |  299.5K | 44 |      4199 |  156.1K |  80 |   44.0 | 38.0 |   3.0 |  88.0 | "31,41,51,61,71,81" | 0:00'38'' | 0:00'30'' |
| MRX160P002 |  160.0 |  19.01% |     20702 | 300.67K | 39 |      5211 | 146.75K |  70 |   44.0 | 36.0 |   3.0 |  88.0 | "31,41,51,61,71,81" | 0:00'39'' | 0:00'30'' |


Table: statMergeAnchors.md

| Name                    | Mapped% | N50Anchor |     Sum |   # | N50Others |    Sum |    # | median |   MAD | lower |  upper | RunTimeAN |
|:------------------------|--------:|----------:|--------:|----:|----------:|-------:|-----:|-------:|------:|------:|-------:|----------:|
| 7_mergeAnchors          |   5.80% |      3708 | 408.46K | 152 |      4036 | 14.26M | 5887 |  440.0 | 238.5 |   3.0 |  880.0 | 0:02'50'' |
| 7_mergeKunitigsAnchors  |  24.68% |      3912 |  357.5K | 125 |      2094 | 13.19M | 6981 |  545.0 | 378.0 |   3.0 | 1090.0 | 0:03'21'' |
| 7_mergeMRTadpoleAnchors |  15.11% |      4164 | 400.64K | 139 |      4080 |  3.98M | 1169 |  475.0 | 364.0 |   3.0 |  950.0 | 0:03'08'' |
| 7_mergeTadpoleAnchors   |  27.16% |      3276 | 320.38K | 118 |      3959 | 12.32M | 4891 |  580.0 | 406.0 |   3.0 | 1160.0 | 0:03'24'' |


Table: statOtherAnchors.md

| Name         | Mapped% | N50Anchor |     Sum |    # | N50Others |     Sum |    # | median |   MAD | lower | upper | RunTimeAN |
|:-------------|--------:|----------:|--------:|-----:|----------:|--------:|-----:|-------:|------:|------:|------:|----------:|
| 8_spades     |  10.86% |      1258 | 839.05K |  638 |      1356 |      3M | 2390 |   23.0 |  10.0 |   3.0 |  46.0 | 0:03'13'' |
| 8_spades_MR  |  21.53% |      1321 |   1.46M | 1076 |      1780 |   3.17M | 2282 |   30.0 |  12.0 |   3.0 |  60.0 | 0:03'08'' |
| 8_megahit    |   8.81% |      1407 | 272.74K |  185 |      9162 | 573.85K |  275 |  111.0 |  84.0 |   3.0 | 222.0 | 0:02'52'' |
| 8_megahit_MR |  25.02% |      1203 | 752.23K |  601 |      1260 |    2.9M | 2372 |   32.0 |  16.0 |   3.0 |  64.0 | 0:03'08'' |
| 8_platanus   |  11.89% |      8044 | 348.81K |   83 |      9311 | 189.92K |   88 |  498.0 | 267.0 |   3.0 | 996.0 | 0:02'45'' |


Table: statFinal

| Name                     |    N50 |       Sum |       # |
|:-------------------------|-------:|----------:|--------:|
| Genome                   | 124033 |    124033 |       1 |
| 7_mergeAnchors.anchors   |   3708 |    408458 |     152 |
| 7_mergeAnchors.others    |   4036 |  14261577 |    5887 |
| anchorLong               |   4175 |    398612 |     141 |
| anchorFill               |  37583 |    411938 |      93 |
| spades.contig            |     98 | 120242351 | 1056261 |
| spades.scaffold          |     98 | 120319869 | 1054687 |
| spades.non-contained     |   1396 |   3835865 |    2584 |
| spades_MR.contig         |   1006 |   9174547 |   12584 |
| spades_MR.scaffold       |   1025 |   9181221 |   12470 |
| spades_MR.non-contained  |   1730 |   4633148 |    2552 |
| megahit.contig           |    374 |  14675664 |   40128 |
| megahit.non-contained    |   2705 |    846589 |     371 |
| megahit_MR.contig        |    611 |  19867743 |   34113 |
| megahit_MR.non-contained |   1297 |   3656452 |    2510 |
| platanus.contig          |     98 |   9716463 |   86683 |
| platanus.scaffold        |    117 |   3559076 |   25664 |
| platanus.non-contained   |  69424 |    538727 |      77 |


# HM340

*Medicago truncatula* HM340

## HM340: download

* Illumina
    * [SRR1524305](https://www.ncbi.nlm.nih.gov/sra/?term=SRR1524305) 33.2 Gb
    * ~85.1x

```bash
mkdir -p ~/data/dna-seq/cpDNA/HM340/1_genome
cd ~/data/dna-seq/cpDNA/HM340/1_genome

ln -fs ../../Medicago/genome.fa genome.fa

mkdir -p ~/data/dna-seq/cpDNA/HM340/2_illumina
cd ~/data/dna-seq/cpDNA/HM340/2_illumina

ln -fs ~/data/dna-seq/cpDNA/Medicago/SRR1524305_1.fastq.gz R1.fq.gz
ln -fs ~/data/dna-seq/cpDNA/Medicago/SRR1524305_2.fastq.gz R2.fq.gz

```


## HM340: template

```bash
WORKING_DIR=${HOME}/data/dna-seq/cpDNA
BASE_NAME=HM340

cd ${WORKING_DIR}/${BASE_NAME}

rm -f *.sh
anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1000000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 170 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160" \
    --tadpole \
    --splitp 100 \
    --statp 2 \
    --fillanchor \
    --redoanchors \
    --parallel 24 \
    --xmx 110g

```

## HM340: run

```bash
WORKING_DIR=${HOME}/data/dna-seq/cpDNA
BASE_NAME=HM340

cd ${WORKING_DIR}/${BASE_NAME}
#rm -fr 4_*/ 6_*/ 7_*/ 8_*/ 2_illumina/trim 2_illumina/mergereads statReads.md 

bash 0_bsub.sh
#bash 0_master.sh

#bash 0_cleanup.sh

```

